editor/components/block-settings-menu/block-settings-menu-plugins-group.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import { flatten, isEmpty, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createSlotFill } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+const { Fill: BlockSettingsMenuPluginsGroup, Slot } = createSlotFill( 'BlockSettingsMenuPluginsGroup' );
+
+const BlockSettingsMenuPluginsGroupSlot = ( { fillProps, selectedBlocks } ) => (
+	<Slot fillProps={ { ...fillProps, selectedBlocks } } >
+		{ ( fills ) => ! isEmpty( fills ) && (
+			<Fragment>
+				<div className="editor-block-settings-menu__separator" />
+				{ fills }
+			</Fragment>
+		) }
+	</Slot>
+);
+
+BlockSettingsMenuPluginsGroup.Slot = withSelect( ( select, { fillProps: { uids } } ) => ( {
+	selectedBlocks: map( flatten( [ uids ] ), ( uid ) => select( 'core/editor' ).getBlockName( uid ) ),
+} ) )( BlockSettingsMenuPluginsGroupSlot );
+
+export default BlockSettingsMenuPluginsGroup;
```

Comment:
Normally we start this extensibility slot names with Plugin e.g: PluginPostPublishPanel, PluginPostStatusInfo etc... So it may make sense to call this component something like PluginBlockSettingsMenu.

Reply:
1237f0b

Reply:
As per https://github.com/WordPress/gutenberg/pull/7895#discussion_r202658539 reverted this change: the external API lives now in `edit-post/block-settings-menu` and can be used as `PluginBlockSettingsMenuItem`.